src/features/private-brand-checks.md

```diff
@@ -0,0 +1,120 @@
+---
+title: 'Private brand checks a.k.a. `#foo in obj`'
+author: 'Marja Hölttä ([@marjakh](https://twitter.com/marjakh))'
+avatars:
+  - 'marja-holtta'
+date: 2021-04-12
+tags:
+  - ECMAScript
+description: 'Private brand checks allow testing for the existence of a private field in an object.'
+tweet: ''
+---
+
+The [`in` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) can be used for testing whether the given object (or any object in its prototype chain) has the given property:
+
+```javascript
+const o1 = {'foo': 0};
+console.log('foo' in o1); // true
+const o2 = {};
+console.log('foo' in o2); // false
+const o3 = Object.create(o1);
+console.log('foo' in o3); // true
+```
+
+The private brand checks feature extends the `in` operator to support [private class fields](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields):
+
+```javascript
+class A {
+  static test(obj) {
+    console.log(#foo in obj);
+  }
+  #foo = 0;
+}
+
+A.test(new A()); // true
+A.test({}); // false
+
+class B {
+  #foo = 0;
+}
+
+A.test(new B()); // false; it's not the same #foo
+```
+
+Since private names are only available inside the class which defines them, the test must also occur inside the class, for example in a method like`static test` above.
+
+Subclass instances inherit private fields from the parent class:
+
+```javascript
+class SubA extends A {};
+A.test(new SubA()); // true
+```
+
+But creating an object with a prototype with `Object.create` or setting the prototype later (via the `__proto__` setter or `Object.setPrototypeOf`) doesn't make private fields accessible:
```

Comment:
Nit: Clarify inheritance:
```suggestion
But objects created with with `Object.create` (or that have the prototype set later via the `__proto__` setter or `Object.setPrototypeOf`) doesn't receive the private fields as own-properties. Because private field lookup only works on own-properties, the `in` operator does not find these inherited fields:
```